pandas/core/internals/construction.py

```diff
@@ -209,7 +210,10 @@ def mgr_to_mgr(mgr, typ: str):
         if isinstance(mgr, ArrayManager):
             new_mgr = mgr
         else:
-            arrays = [mgr.iget_values(i).copy() for i in range(len(mgr.axes[0]))]
+            if copy:
+                arrays = [mgr.iget_values(i).copy() for i in range(len(mgr.axes[0]))]
+            else:
+                arrays = [mgr.iget_values(i) for i in range(len(mgr.axes[0]))]
```

Comment:
can do this unconditionally and then `if copy: arrays = [x.copy() for x in arrays]`